src/platform/qpg6100/Logging.cpp

```diff
@@ -62,6 +62,9 @@ void LogV(uint8_t module, uint8_t category, const char * msg, va_list v)
         char formattedMsg[CHIP_DEVICE_CONFIG_LOG_MESSAGE_MAX_SIZE];
         size_t prefixLen;
 
+        constexpr size_t maxPrefixLen = chip::Logging::kMaxModuleNameLen + 3 + 3;
```

Comment:
This does not look big enough.  We need 3 chars for "[E]" (or D or P), then `kMaxModuleNameLen` chars for the `GetModuleName`, 3 more chars for the `[` `]` around the module name and the following space.

But we also need at least one more byte for the null-terminator.  Otherwise the `vsnprintf` will leave the buffer unterminated, and when we do the `qvCHIP_Printf` we will start reading random out-of-buffer memory.

Reply:
I'm expecting the prefix to be = 3 + 1+ kMaxModuleNameLen + 1+ 1
For example = "[E][someModule] "

For the extra char for the null-terminator - I was relying on vsnprintf to take care of that in essence.
https://stackoverflow.com/questions/7706936/is-snprintf-always-null-terminating
If there is an expectation vsnprintf is not NULL terminating the buffer, we can apply a -1 to the vnsprintf's call

Reply:
vsnprintf always null-terminates so long as it hasn't returned a negative number.  However, it can truncate to do so.  So obviously it's better to account for the null-character.

Reply:
Ah, I misread the static assert.  It has `>`, not `>=`, which is what makes this work: `sizeof(formattedMsg) - prefixLen` can't be zero.  OK, great.  :)